serverless-functions/src/functions/features/callback-and-voicemail/studio/wait-experience.protected.js

```diff
@@ -109,32 +115,42 @@ async function handleCallbackOrVoicemailSelected(context, isVoicemail, callSid,
   const domain = `https://${context.DOMAIN_NAME}`;
 
   const cancelReason = isVoicemail ? 'Opted to leave a voicemail' : 'Opted to request a callback';
-  const mode = isVoicemail ? 'record-voicemail' : 'submit-callback';
+  const mode = isVoicemail ? 'record-voicemail' : 'handle-callback-choice';
 
   const task = await fetchTask(context, taskSid);
 
   // Redirect Call to callback or voicemail logic. We need to update the call with a new TwiML URL vs using twiml.redirect() - since
   // we are still in the waitUrl TwiML execution - and it's not possible to use the <Record> verb in here. We piggyback on the same approach for callbacks,
   // though technically these could be handled entirely in the waitUrl TwiML
```

Comment:
In my opinion, now that the voicemail and callback logic are separate, this function doesn't make sense to handle both any more. Maybe it makes sense to split this function up?